modules/amf-event-monitor-web/src/main/java/com/liferay/registration/web/portlet/util/AuditEventUtil.java

```diff
@@ -0,0 +1,198 @@
+import com.liferay.portal.security.audit.AuditEventManager;
+import com.liferay.portal.security.audit.AuditEvent;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component(immediate = true, service = {})
+public class AuditEventUtil {
+
+    public static AuditEvent fetchAuditEvent(long auditEventId) {
+        return  _auditEventManager.fetchAuditEvent(auditEventId);
+    }
+
+    public static List<AuditEvent> getAuditEvents(long companyId, int start, int end,
+                                                  OrderByComparator orderByComparator) throws JSONException {
+
+        return editAssignUserInformationEventList(_auditEventManager.getAuditEvents(
+                companyId, start, end, orderByComparator));
+    }
+
+    public static List<AuditEvent> getAuditEvents(
+            long companyId, long userId, String userName, Date createDateGT,
+            Date createDateLT, String eventType, String className, String classPK,
+            String clientHost, String clientIP, String serverName, int serverPort,
+            String sessionID, boolean andSearch, int start, int end,
+            OrderByComparator orderByComparator) {
+
+        return _auditEventManager.getAuditEvents(
+                companyId, userId, userName, createDateGT, createDateLT, eventType,
+                className, classPK, clientHost, clientIP, serverName, serverPort,
+                sessionID, andSearch, start, end, orderByComparator);
+    }
+
+    public static int getAuditEventsCount(long companyId) {
+        return _auditEventManager.getAuditEventsCount(companyId);
+    }
+
+    public static int getAuditEventsCountByType(long companyId, String eventType, OrderByComparator orderByComparator) throws JSONException {
+        List<AuditEvent> auditEventList = getAuditEvents(companyId,0,getAuditEventsCount(companyId),orderByComparator);
+        return (int)auditEventList.stream().filter(auditEvent -> auditEvent.getEventType().equals(eventType)).count();
+    }
+
+    public static int getAuditEventsCountByUserId(long companyId, long userId, OrderByComparator orderByComparator) throws JSONException {
+        List<AuditEvent> auditEventList = getAuditEvents(companyId,0,getAuditEventsCount(companyId),orderByComparator);
+        return (int)auditEventList.stream().filter(auditEvent -> auditEvent.getUserId()==userId).count();
+    }
+
+    public static int getAuditEventsCountByTypeAndUserId(long companyId, String eventType, long userId, OrderByComparator orderByComparator) throws JSONException {
+        List<AuditEvent> auditEventList = getAuditEvents(companyId,0,getAuditEventsCount(companyId),orderByComparator);
+        return (int)auditEventList.stream().filter(auditEvent -> auditEvent.getEventType().equals(eventType)&& auditEvent.getUserId()==userId).count();
+    }
+
```

Comment:
I think it should use a custom Finder to implement this.
Because the current implementation could cause a performance problem.
https://help.liferay.com/hc/en-us/articles/360030614252-Defining-a-Custom-Finder-Method